include/firefly/intel64/cpu/cpu.hpp

```diff
@@ -9,6 +9,34 @@
 // There is no smp yet, but I'd prefer to structure
 // everything in a way so that it's easy to get it up and running.
 namespace firefly::kernel {
+struct __attribute__((packed)) InterruptStack {
+    int64_t r15;
+    int64_t r14;
+    int64_t r13;
+    int64_t r12;
+    int64_t r11;
+    int64_t r10;
+    int64_t r9;
+    int64_t r8;
+    int64_t rdi;
+    int64_t rsi;
+    int64_t rbp;
+    int64_t rdx;
+    int64_t rcx;
+    int64_t rbx;
+    int64_t rax;
+
+    int64_t int_no;
+    int64_t err;
+
+    int64_t rip;
+    int64_t cs;
+    int64_t rflags;
+    int64_t rsp;
+    int64_t ss;
+};
+
+typedef InterruptStack Registers;
```

Comment:
This is confusing.

`Registers` gives the reader the impression that it is a struct containing General purpose registers, but in reality it's an interrupt frame.

InterruptStack might also imply that it is an actual stack, i.e. IST.
It would be best to rename that as well.

Could be `Registers` -> `ContextRegisters` and `InterruptStack` -> `InterruptFrame` or something.